File: designPattern/visitor/review/Directory.java
import java.util.ArrayList;
import java.util.Iterator;

public class Directory extends Entry{
	String name;
	ArrayList dirs = new ArrayList();
	public Directory(String name){
		this.name = name;
	}
	public String getName(){
		return name;
	}
	public int getSize(){
		Iterator it = iterator();
		int size = 0;
		while(it.hasNext()){
			Entry entry = (Entry)it.next();
			size+=entry.getSize();
		}	
		return size;
	}
	public Iterator iterator(){
		return dirs.iterator();
	}

	public void add(Entry entry){
		dirs.add(entry);
	}
	public void accept(Visitor v){
		v.visit(this);
	}
}

File: designPattern/visitor/pythonbasic/Directory.py
from Element import Element
from Entry import Entry


class Directory(Element, Entry):
    def __init__(self, name):
        self.name = name
        self.documents = []

    def getName(self):
        return self.name

    def getSize(self):
        size = 0
        for docs in self.documents:
            size += docs.getSize()
        return size

    def add(self, file_or_directory):
        self.documents.append(file_or_directory)

    def accept(self, visitor):
        visitor.visitDirectory(self)